blog/content/post/2020-08-05-PCIe-features.md

```diff
@@ -0,0 +1,177 @@
+---
+title: What features PCIe has?
+abstract: 'Introductory blog post to PCIe features.
+          In this article you can read what PCIe capability is
+          and see examples of such capabilities.'
+cover: /covers/PCIExpress.jpg
+author: marek.kasiewicz
+layout: post
+published: true
+date: 2020-08-05
+archives: "2020"
+
+tags:
+  - Firmware
+categories:
+  - Firmware
+  - IoT
+  - Miscellaneous
+  - OS Dev
+  - App Dev
+  - Security
+  - Manufacturing
+
+---
+
+## What this blog is about
+
+Goal of this post is to introduce readers to the concept of PCIe capabilities.
+It briefly describes what capability is and how it works. It also lists some
+example capability described in PCIe specification.
+
+## Introduction
+
+Every device in the PCIe tree has at least one function. Each function has its
+configuration space. Configuration space is device registers mapped to either
```

Comment:
What is the configuration space for? What can we do with it? It is okay to list raw facts about PCIe devices, but leaving them without explanation dramatically reduces the readability of the post.